src/main/java/io/wisetime/connector/jira/JiraConnector.java

```diff
@@ -263,4 +271,19 @@ private String tagUpsertPath() {
                 .toArray(String[]::new)
         ).orElse(ArrayUtils.toArray());
   }
+
+  private Optional<String> getJiraUser(User user) {
+    if (StringUtils.isNotBlank(user.getExternalId())) {
+      return Optional.ofNullable(
+          jiraDao.findUserByUsername(user.getExternalId())
+            .orElseGet(() ->
+                EMAIL_REGEX.matcher(user.getExternalId()).find()
+                    ? jiraDao.findUserByEmail(user.getExternalId()).orElse(null)
+                    : null
```

Comment:
The code gets unnecessary complicated. Would you refactor it using more variables for better readability?

Reply:
I have updated the PR. Thanks